### DISCUSSION TITLE
UserMenuItem() label translation not working

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6367] [Help]
I create user menu like this [https://filamentphp.com/docs/2.x/admin/navigation#customizing-the-user-menu](https://filamentphp.com/docs/2.x/admin/navigation#customizing-the-user-menu) and change label with translation key `->label(trans('message.user_account'))`

But it only display English language.. it cannot change to another language..

**Edited**
It display `config.app.locale` only, after switch language it cannot change

Version
`"filament/filament": "^2.0",`
`"laravel/framework": "^10.0",`

**Comment by danharrin** [comment]
Do you have middleware which sets the locale? This is probably causing issues as the locale has not been set by the time the menu item is registered.

You probably need to create a new middleware after the locale middleware to register these items inside.

**Reply by valpuia** [reply]
No, I don't have any middleware, I create `message.php` file in `lang` folder and store `key=>value` of locale.. and change locale using `session`

**Reply by danharrin** [reply]
Where in your code do you set the locale

**Reply by valpuia** [reply]
I use this package [https://filamentphp.com/plugins/language-switch](https://filamentphp.com/plugins/language-switch) but in this code what it does is below line (when click language list in dropdown)

```
public function changeLocale($locale)
{
    session()->put('locale', $locale);

    $this->redirect(request()->header('Referer'));
}
```

**Reply by danharrin** [reply]
That package uses middleware. You need to open an issue on `bezhanSalleh/filament-language-switch` asking how to wait until the locale is set before registering user menu items.

**Reply by ArtrenH** [reply]
I have the same problem but I'm using this custom middleware for every request:
```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\App;
use Symfony\Component\HttpFoundation\Response;


class SetBrowserLocale
{
    /**
     * Handle an incoming request.
     */
    public function handle(Request $request, Closure $next): Response
    {
        //get langs
        $lang = '';
        if (array_key_exists('HTTP_ACCEPT_LANGUAGE', $_SERVER)) {
            $lang = $_SERVER['HTTP_ACCEPT_LANGUAGE'];
        } // bulletproof if api-client request like a browser

        if (strlen($lang) > 1) {  // bulletproof if api-client request like a browser

            //chrome sends de-DE instead of de, so we ignore the second part:
            $lang = substr($lang, 0, 2);

            //set locale
            App::setLocale($lang);
        }

        return $next($request);
    }
}

```
What could I do to translate the menu items?

**Reply by ArtrenH** [reply]
I solved my issue with using
```php
->label(
    fn () => __('Manual')
)
``` instead of 
```php
->label(__('Manual'))
```
but maybe there is a more elegant approach
